docs/contributing/03-stories/story-32.md

```diff
@@ -0,0 +1,107 @@
+  validate?(config: any): any;
+}
+```
+
+As he understands it, Wing comes with a set of built-in platforms `tf-aws`, `tf-azure`, `tf-gcp`, and `sim`. These platforms define how the resources are translated into
+the underlying infrastructure and inflight clients. He knows his company is an AWS shop and many in the company have extensive experience with `Terraform` so the `tf-aws` platform sounds like a great
+place to start.
+
+He has determined that he can create a new platform lovingly dubbed `MarcusPlatform` which should be able to just extend the `tf-aws` platform and override a few methods to
+enforce the company standards.
+
+He begins by creating his platform in a new npm project:
+
+```typescript
+// marcus-platform.ts
+import { Construct } from "constructs";
+import { IPlatform, TfAWS } from "wingsdk/platform";
+
+export class MarcusPlatform extends TfAWS.Platform {
+  newInstance(type: string, scope: Construct, id: string, props: any): any {
+    if (type === "bucket") {
+      // return custom bucket
+      return new Bucket(scope, id, props);
+    }
+
+    return super.newInstance(type, scope, id, props);
+  }
+}
+
+// Custom Bucket class
+export class Bucket extends TfAWS.Bucket {
+  constructor(scope: Construct, id: string, props: any) {
+    super(scope, id, props);
+
+    // enable bucket versioning
+    const bucket = new S3Bucket(scope, name, {
+      versioning: {enabled: true},
+    });
+  }
+}
+```
+
+As Marcus continues to change the resources to implement his companies standards he recalls that all IAM roles in his company must have a permission boundary set. 
+
+He recalls the `preSynth` method from the `IPlatform` interface which will allow him to traverse the Construct tree before any synthesis occurs. Meaning he can apply the permission boundary to all IAM roles before the code is compiled.
+
+
+He is able to just add the following method to his platform
+```typescript
+  // .. code omitted for brevity
+  export class MarcusPlatform extends TfAWS.Platform {
+    validate(app: Construct): void {
+      // apply the Permission boundary aspect
+      cdktf.Aspects.of(app).add(new PermissionBoundaryAspect("some-boundary-arn"));
+    }
+  }
+  // .. code omitted for brevity
+  class PermissionBoundaryAspect {
+    constructor(permissionBoundaryArn: string) {
+      this.permissionBoundaryArn = permissionBoundaryArn;
+    }
+
+    visit(node: any) {
+      if (node.terraformResourceType === "aws_iam_role") {
+        node.permissionsBoundary = this.permissionBoundaryArn;
+      }
+    }
+  }
+```
+
+Finally he publishes his platform to his companies private npm registry. 
+
+He is now able to share with his internal teams that if they want to compile and deploy their code to AWS they can use the `MarcusPlatform` platform as such:
+
+```bash
+# install the platform
+npm install @company/marcus-wing-platform
+# compile with the platform
+wing compile app.main.w --platform tf-aws --platform MarcusPlatform
```

Comment:
Since the new platform extends from `tfaws.Platform`, I don't think we need to compose them in the command line:

```suggestion
wing compile app.main.w --platform @company/marcus-wing-platform.Platform
```

I think `wing compile` will need to know the module, no?

Reply:
Im not entirely sure whats best here. I was kind of thinking the convention can be that there should always be a 
`export const Platform = SomeCustomPlatform` so Wing can always just assume the module is called Platform

Reply:
We can have that as a default (so I can just do `-p @company/marcus-wing-platform`) but also specify an explicit class